MessageViewController/MessageAutocompleteController.swift

```diff
@@ -156,13 +199,46 @@ public final class MessageAutocompleteController: MessageTextViewListener {
             else { return }
         keyboardHeight = keyboardFrame.height
     }
+    
+    /// Ensures new text typed is not styled
+    ///
+    /// - Parameter textView: The `UITextView` to apply `typingTextAttributes` to
+    internal func preserveTypingAttributes(for textView: UITextView) {
+        
+        var typingAttributes = [String: Any]()
+        typingTextAttributes.forEach { typingAttributes[$0.key.rawValue] = $0.value }
+        textView.typingAttributes = typingAttributes
+    }
 
     // MARK: MessageTextViewListener
 
     public func didChangeSelection(textView: MessageTextView) {
         check()
     }
 
-    public func didChange(textView: MessageTextView) {}
+    public func didChange(textView: MessageTextView) {
+        preserveTypingAttributes(for: textView)
+    }
+    
+    public func willChangeRange(textView: MessageTextView, to range: NSRange) {
+        
+        if range.length > 0, range.lowerBound < textView.selectedRange.lowerBound {
+            // Backspace/removing text
+            let attribute = textView.attributedText.attributes(at: range.lowerBound, longestEffectiveRange: nil, in: range).filter { return $0.key == NSAttributedAutocompleteKey }
+            let isAutocompleted = (attribute[NSAttributedAutocompleteKey] as? Bool ?? false) == true
+            if isAutocompleted {
```

Comment:
merge `isAutocompleted` into the if-statement? variable isn't used anywhere else

Reply:
I feel like it improves readability here as it's a quite complicatedly (is that a word?) computed `Bool`.